### PR TITLE
Add `TableInfoPopover` and add it to Table cards on the `/browse` route

### PR DESCRIPTION
**Pull request by daltojohnso**
#18738

This PR adds a "TableInfoPopover" to table cards found on the `/browse` route, navigated to via the "Browse Data" button in the header.

**Testing**
1. Navigate to the `/browse` route and select a database
2. Hover over table cards to see the popover
3. On second hover, you shouldn't see a loading state for field/fk metadata
4. Click on "connected tables" links to be navigated to a simple question view of the specified table

Screen Shot 2021-12-13 at 1 37 01 PM

https://user-images.githubusercontent.com/13057258/146834300-28b23441-6aa2-488c-a060-0f3f5f7d444d.mov

### CODE REVIEW

File: frontend/src/metabase/components/MetadataInfo/TableInfoPopover/TableInfoPopover.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import TippyPopver from "metabase/components/Popover/TippyPopover";
```
Comment (mazameli) [inline frontend/src/metabase/components/MetadataInfo/TableInfoPopover/TableInfoPopover.jsx:4]: …did you name this after Clippy? 😂 📎

**Comment by mazameli** [comment]
It looks like the first time Tippy appears for a given table, it enters in abruptly; but then subsequent times it loads in nice and smooth. In this recording, the first couple of tables I'm hovering over have already displayed the popover once before so they appear nicely, and then you'll see some abrupt ones after that: https://www.loom.com/share/c2318814b7734ec2b4c29a2d983cf4de

**Comment by mazameli** [comment]
FYI I'm gonna push up a tiny commit to slightly decrease the delay for displaying these; feels too long to me in practice.

**Comment by mazameli** [comment]
Clicking on a liked table seems to take me to the Data Reference instead of to the Simple mode view as the spec says. ("Items in this list should be clickable, taking the user to a new Simple Mode question using the clicked table as starting data.") I can't recall for sure if this ways a deliberate change we made along the way, but I don't believe so.

**Comment by mazameli** [comment]
Those two things I commented on appear to be the only issues I can find. 👏🏼

**Comment by daltojohnso** [comment]
@mazameli -- the initial delay are the requests to make to populate the column count and the connected columns. I'll add a loading state for those bits so that we can show the popover immediately.

**Comment by daltojohnso** [comment]
@mazameli -- I've fixed those two problems. LMK what you think. Can definitely add a more complex loading placeholder if desired.

**Comment by mazameli** [comment]
This looks great!

File: frontend/src/metabase/components/MetadataInfo/TableInfoPopover/TableInfoPopover.styled.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import TableInfo from "metabase/components/MetadataInfo/TableInfo";
+
+type TableInfoProps = {
+  tableId: number;
+};
+
+export const WidthBoundTableInfo = styled(TableInfo)<TableInfoProps>`
+  max-width: 300px;
+  min-width: 300px;
```
Comment (gusaiani) [inline frontend/src/metabase/components/MetadataInfo/TableInfoPopover/TableInfoPopover.styled.tsx:11]: Could this simple be `width: 300px`? 🤔

File: frontend/test/metabase/scenarios/onboarding/reference/databases.cy.spec.js [hunk 3]
```diff
@@ -77,6 +77,25 @@ describe("scenarios > reference > databases", () => {
       checkQuestionSourceDatabasesOrder("Native query");
     });
   });
+
+  it("should show a popover with metadata related to the table", () => {
+    cy.visit("/browse/1");
+    cy.findByText("Products").trigger("mouseenter");
+
+    popover().within(() => {
+      // check that is has shows the table's description
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/onboarding/reference/databases.cy.spec.js:86]:
```suggestion
      // check that is has shown the table's description
```

?

File: frontend/test/metabase/scenarios/onboarding/reference/databases.cy.spec.js [hunk 4]
```diff
@@ -77,6 +77,25 @@ describe("scenarios > reference > databases", () => {
...
+      cy.contains("This is our product catalog");
+
+      // check that it shows the number of columns in the table
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/onboarding/reference/databases.cy.spec.js:89]:
```suggestion
      // check that it shown the number of columns in the table
```

?

**Review by gusaiani — APPROVED** [review]
👍, just a few very minor comments inlined.

File: frontend/src/metabase/components/MetadataInfo/TableInfoPopover/TableInfoPopover.styled.tsx (hunk 2 above)
Comment (daltojohnso) [reply frontend/src/metabase/components/MetadataInfo/TableInfoPopover/TableInfoPopover.styled.tsx]: true; fixed

File: frontend/test/metabase/scenarios/onboarding/reference/databases.cy.spec.js (hunk 3 above)
Comment (daltojohnso) [reply frontend/test/metabase/scenarios/onboarding/reference/databases.cy.spec.js]: besides the grammar error, the sentence felt a bit verbose, so i chopped it down a bit

**Pull request merged**